src/ui/settings.ts

```diff
@@ -238,6 +239,59 @@ export class SettingTab extends PluginSettingTab {
     });
   }
 
+  generateCustomRegexReplacementSettings(tabName: string, containerEl: HTMLElement): void {
+    containerEl.createEl(Platform.isMobile ? 'h4' : 'h3', {text: 'Custom Regex Replacement'});
+
+    new Setting(containerEl)
+        .addButton((cb)=>{
+          cb.setButtonText('Add new regex')
+              .setCta()
+              .onClick(()=>{
+                this.plugin.settings.customRegexs.push({find: '', replace: ''});
+                this.plugin.saveSettings();
+                this.display();
+                const customRegexInputBox = document.getElementsByClassName('linter-custom-regex-replacement');
+                // @ts-ignore
+                customRegexInputBox[customRegexInputBox.length-1].focus();
+              });
+        });
+
+    let searchInfo = '';
+    this.plugin.settings.customRegexs.forEach((regex, index) => {
+      searchInfo += regex.find + ' ';
+      searchInfo += regex.replace + ' ';
+      new Setting(containerEl)
+          .addText((cb) => {
+            cb.setPlaceholder('regex to find')
+                .setValue(regex.find)
+                .onChange((value) => {
+                  this.plugin.settings.customRegexs[index].find = value;
+                  this.plugin.saveSettings();
+                });
+            cb.inputEl.setAttr('inputIndex', index);
+            cb.inputEl.addClass('linter-custom-regex-replacement');
+          })
+          .addText((cb) => {
+            cb.setPlaceholder('regex to replace')
+                .setValue(regex.replace)
+                .onChange((value) => {
+                  this.plugin.settings.customRegexs[index].replace = value;
+                  this.plugin.saveSettings();
+                });
+          }).addExtraButton((cb)=>{
+            cb.setIcon('cross')
+                .setTooltip('Delete')
+                .onClick(()=>{
+                  this.plugin.settings.customRegexs.splice(index, 1);
+                  this.plugin.saveSettings();
+                  this.display();
+                });
+          });
+    });
+
+    this.addSettingToMasterSettingsList(tabName, containerEl as HTMLDivElement, tabName.toLowerCase(), searchInfo);
```

Comment:
The search info is actually the name and description of the rule itself. So you would likely just have the name of the setting placed here for search info: "Custom Regex Replacement".
A description is welcome as well though.